Builder/ElasticDocumentBuilder.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Markup\NeedleBundle\Builder;
+
+class ElasticDocumentBuilder implements ElasticDocumentInterface
+{
+    /**
+     * @var array
+     */
+    protected $data = [];
+    /**
+     * @var string
+     */
+    private $index;
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @param string $index
+     * @param string $type
+     * @param null|string $id
+     */
+    public function __construct(string $index, string $type, ?string $id)
+    {
+        $this->data = [
+            'index' => $index,
+            'type' => $type
+        ];
+
+        if ($id) {
+            $this->data['id'] = $id;
+        }
+
+        $this->index = $index;
+        $this->type = $type;
+    }
+
+    /**
+     * @param array $inputData
+     */
+    public function addData(array $inputData)
+    {
+        if (isset($this->data['body'])) {
+            $this->data['body'] = array_merge($this->data['body'], $inputData);
+        } else {
+            $this->data['body'] = $inputData;
```

Comment:
Seems to me we could just do `$this->data['body'] = []` in the construct then remove the need for condition completely.